### PR TITLE
Config parser

### LINKED ISSUE #5: Issue with use of configargparse and -h
When `platalea.config` is imported in a script, it catches the `-h` (help) parameter before the script's specific parameters are added. As an example, `python utils/evaluate_asr.py -h` doesn't show the script's parameters (`path`, `-b`), only the global ones (e.g. `data_root`, `meta`). Importing config after the script has parsed the parameters would result in the opposite behavior.

@egpbos, any idea how this could be handled better?

### PR DESCRIPTION
**Pull request by egpbos**
Fixes #5.

It implements a class that has default arguments available by default and only parses when you call parse() on it.

By default, displaying help is disabled, but it can be reenabled after all arguments have been added from an experiment script. It's not actually currently working properly, so have to fix still.

It is working from basic-default, still have to fix all the other experiment scripts to work.

I removed the default values from the platalea.dataset code, which now I think about it probably removes the need for this entire class... Most of the rewriting will actually be to pass through the now necessary default arguments, because they are no longer embedded in dataset. So, may remove the class later in this PR.

In summary: do not merge yet ;)

### CODE REVIEW

**Comment by egpbos** [comment]
Ok, so I did a relatively major refactoring of the distribution (note, terminology (this always trips me up): module = file, package = directory, distribution = entire repo containing now four packages):

- Moved utils and experiments directories under platalea directory and made them installable subpackages.
- Also made the audio directory an installable subpackage (see setup.py).
- Removed all references to `config` module in platalea package itself.
- Moved all the run.py files to be directly in the experiments folder with the name of their former subdirectories, so e.g. `basic-default/run.py` is now just `basic_default.py`. Also note the change from dash to underscore; this allows us to import the files as modules, because dashes are not valid in module names.

After installing, you can now run experiments without knowing the exact file location, e.g.:

```py
python -m platalea.experiments.basic_default
```

Not all of this was strictly necessary, but especially making the `experiments` folder into an actual package solved the problem that I wanted to move the `config` module to `experiments` (where I think it rightly belongs, especially now that I removed all references to it from the main `platalea` package itself), but that I couldn't reliably import it from there, because experiments was just a folder. Now we can just do `import platalea.experiments.config` instead of having to do some complex magic with absolute paths and relative imports. I guess an alternative would have been to just make a separate config package/module and keep experiments where they were, but I think in any case this is cleaner. Let me know what you think.

The trans, vgs and vgs-asr experiments have not been transferred and will not be installed in the current version, but since they weren't updated to use the configargparse stuff, I wasn't sure whether they were still relevant at all. In any case, we can do those later if necessary.

At this point the help print is still broken, but the other issues mentioned above (fixing remaining experiments and the remark about not using the class at all) are fixed. So to do still:

- [x] Fix help print
- [x] Fix CI build

**Comment by egpbos** [comment]
Ok, fixed the above two items, but turns out I hadn't pulled the latest master, so I now need to ~rebase~ merge with master.

**Comment by egpbos** [comment]
Wow, this is not a trivial merge, you changed a lot :) I guess next time I should review newest PRs first instead of starting at the oldest issue...

**Comment by egpbos** [comment]
Ok, I'll leave this PR for now and start a new branch off master, probably easier, because too many files were moved or deleted, in both our branches :D
